packages/ember-glimmer/tests/integration/syntax/each-in-test.js

```diff
@@ -0,0 +1,134 @@
+import { set } from 'ember-metal/property_set';
+import { strip } from '../../utils/abstract-test-case';
+import { moduleFor, RenderingTest } from '../../utils/test-case';
+
+// TODO(mmun): Extend BasicConditionalsTest for a more robust test matrix.
+
+moduleFor('@htmlbars Syntax test: {{#each-in}}', class extends RenderingTest {
+
+  [`@test it repeats the given block for each item in the hash`]() {
+    this.render(strip`
+      <ul>
+        {{#each-in categories as |category count|}}
+          <li>{{category}}: {{count}}</li>
+        {{/each-in}}
+      </ul>
+    `, {
+      categories: {
+        'Smartphones': 8203,
+        'JavaScript Frameworks': Infinity
+      }
+    });
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Smartphones: 8203</li>
+        <li>JavaScript Frameworks: Infinity</li>
+      </ul>
+    `);
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'categories', {
+      'Smartphones': 100
+    }));
+
+    this.assertHTML(`<ul><li>Smartphones: 100</li></ul>`);
+
+    this.assertStableRerender();
+
+    this.runTask(() => set(this.context, 'categories', {
+      'Programming Languages': 199303,
+      'Good Programming Languages': 123,
+      'Bad Programming Languages': 456
+    }));
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Programming Languages: 199303</li>
+        <li>Good Programming Languages: 123</li>
+        <li>Bad Programming Languages: 456</li>
+      </ul>
+    `);
+
+    this.assertStableRerender();
+  }
+
+  [`@test it only iterates over an object's own properties`]() {
+    let protoCategories = {
+      'Smartphones': 8203,
+      'JavaScript Frameworks': Infinity
+    };
+
+    let categories = Object.create(protoCategories);
+    categories['Televisions'] = 183;
+    categories['Alarm Clocks'] = 999;
+
+    this.render(strip`
+      <ul>
+        {{#each-in categories as |category count|}}
+          <li>{{category}}: {{count}}</li>
+        {{/each-in}}
+      </ul>
+    `, { categories });
+
+    this.assertHTML(strip`
+      <ul>
+        <li>Televisions: 183</li>
+        <li>Alarm Clocks: 999</li>
+      </ul>
+    `);
+
+    this.assertStableRerender();
+  }
```

Comment:
Can we merge this test into the previous test using the "I-N-U-R" pattern? (I agree that in this case the existing test already covered the same things we want to cover with that pattern, but a lot of existing tests doesn't do it. It seems good to have one pattern that we can bring everywhere and we are guaranteed to get good coverage every time.)